### ISSUE TITLE
Track 3: two small logging gaps that make a record hard to verify from its logfile

### ISSUE THREAD

**Issue by view321** [KellerJordan/modded-nanogpt#355]
While auditing the track-3 end-of-training readout lineage (`#38` → `#42` → `#43` →
`#45` → `#46`) by reading the rule-4 embedded sources, I hit two places where a
logfile does not describe its own run. Both are one-line fixes and neither is
evidence of anything wrong — in both cases another record in the same lineage
already does the right thing, so this is really a request to make the odd one out
match its siblings.

Happy to send a PR for either or both if useful.

*(Numbers like `#46` below are **row numbers in the track-3 results table**, not
PR/issue numbers. Record `#45` is PR #325; record `#46` is PR #328.)*

---

## 1. Record `#46`'s logs do not record which of its levers were enabled

Four records in this lineage embed an **env-gated research trainer** rather than a
hardcoded script. That satisfies rule 4 literally — the code is in the logfile —
but the behaviour is then selected by environment variables, so the log only
describes its own run if it prints them.

**Three of the four do, thoroughly:**

| record | `Using …` config lines in the log | covers its readout params? |
|---|---:|---|
| `#38` | 130 | yes (`ref_extrap_*`, `tail_vel_*`, …) |
| `#43` | 110 | yes (`trail_delta_*`, `phase_readout_*`, `final_readout_*`, …) |
| `#45` | 59 | yes (`tail_ema_taus/start_step/apply_lambda`, …) |
| **`#46`** | **18** | **no** |

`#46`'s embedded trainer defaults its three headline levers to **off**:

```python
ROWFLOOR    = int(os.environ.get("ROWFLOOR", 0))
CWD         = float(os.environ.get("CWD", 0.0))
TAILEMA_TAU = float(os.environ.get("TAILEMA_TAU", 0))
```

and none of `ROWFLOOR`, `CWD`, `TAILEMA_TAU` appears among those 18 lines. So the
logfile does not record that RowFloor, Cautious Weight Decay or Tail-EMA were
enabled for the run it contains.

The cleanest way to see the gap is to compare against `#45`, which is env-gated in
exactly the same style but whose **defaults reproduce its own record**:

```python
TAIL_EMA_TAUS        = ... os.environ.get("TAIL_EMA_TAUS", "150")
TAIL_EMA_START_STEP  = int(os.environ.get("TAIL_EMA_START_STEP", "2000"))
TAIL_EMA_APPLY_STEP  = int(os.environ.get("TAIL_EMA_APPLY_STEP", "2720"))
TAIL_EMA_APPLY_TAU   = float(os.environ.get("TAIL_EMA_APPLY_TAU", "150"))
TAIL_EMA_APPLY_LAMBDA= float(os.environ.get("TAIL_EMA_APPLY_LAMBDA", "0.6"))
```

which is exactly what its log reports (`tail_ema applied step:2720 tau:150
lambda:0.6 start_step:2000`). So:

- run `#45`'s embedded source with **no environment set** → you reproduce `#45`;
- run `#46`'s embedded source with **no environment set** → you get the `#44` base,
  not `#46`.

That difference is the whole of this request.

**To be clear about what this is and isn't.** The submitters document the
required environment in their README, and they also ship a hardcoded
`train_gpt_cwd_SOTA.py` alongside the logs — which is the Guidelines followed to
the letter. This is a transparency gap in one artifact, not a correctness
problem, and I have no reason to doubt the result.

But it does have a concrete consequence, and the three levers differ in how
recoverable they are:

- **Tail-EMA is inferable.** `val_ema_loss` is printed unconditionally and
  defaults to `val_loss` when the EMA is off, so a reader who notices the two
  columns diverging after step 2400 can conclude it was on.
- **RowFloor and CWD leave no trace at all.** They change training, so they only
  move `val_loss` itself, and there is nothing inside the log to compare that
  against.

So a `#46` record log and a `#44`-base control log would be identical in their code
section, identical in their config section, and differ only in the value of
`val_loss` — which is precisely the number at stake in the "Skeptical results"
scenario the README describes.

### Suggested fix

One line before training starts, dumping every gating variable the trainer reads
— exactly what `#38`/`#43`/`#45` already emit. Something like:

```python
for k in sorted(GATED_ENV_VARS):
    print0(f"Using {k.lower()}={os.environ.get(k, '<default>')}")
```

### Possibly also worth a Guidelines line

The Guidelines currently say:

> In particular, hardcoded hyperparameters are to be preferred as compared to
> command line arguments.

Environment variables are strictly worse than command-line arguments for this
purpose: a command line at least tends to survive in shell history or a `run.sh`,
whereas an env var leaves no trace in any artifact. Extending that sentence to
mention environment variables — or asking that gating values be printed — would
close the gap for future submissions rather than just this one.

---

## 2. Record `#38`'s accepted val line is readout-transformed but not marked

`#38` applies a reference-interpolation readout at validation
(`γ = −0.075` toward the step-2375 weights). Its config sets

```
Using ref_extrap_capture_step=2375
Using ref_extrap_apply_start_step=2900
Using ref_extrap_gamma=-0.075
```

and the accepted step is **2900**, so the readout fires at **exactly one
validation point in the whole run — the accepted one**. The log line for it reads:

```
step:2900/2900 val_loss:3.27803 train_time:4798.129s step_avg:1654.53ms
```

Nothing there indicates the number is readout-transformed. Because the readout
fires only at 2900, the log also contains no un-transformed loss at that step to
compare against, so a reader diffing `#38` against a baseline log has no way to
notice.

Record `#42` handles the *same kind of readout* correctly, emitting a suffix:

```
step:2860/2900 val_loss:3.27837 ... ref_gamma:-0.075 ref_step:2375
```

(`#38` and `#42` use logically identical readout code with the same `γ = −0.075` and
the same capture step 2375, so the two logs are directly comparable — one marks
the line, the other doesn't.)

### Suggested fix

Emit `#42`'s suffix whenever the applied γ is non-zero. `#38`'s own source already
has the pieces; it just doesn't append them to the validation print.

---

## Why bother

The README's "Skeptical results" section invites challenges to old results and
treats them as first-class. That only works if a logfile is checkable on its own
terms. Both changes above are small, both have an in-repo precedent in the same
record lineage, and together they'd make the whole readout lineage auditable by
reading — which is how I found everything in the companion discussion, at zero
GPU cost.

Companion discussion (the science; this issue is just the housekeeping):
https://github.com/KellerJordan/modded-nanogpt/discussions/354

Found while auditing with [Grad](https://github.com/view321/Grad), a free
open-source research harness running Claude. The audit above is all source
reading and log parsing — no GPU, and the probes are in that repo.
